src/main/java/sg/ncl/MainController.java

```diff
@@ -2744,6 +2746,101 @@ public String adminTeamUsage(Model model,
         return "usage_statistics";
     }
 
+    @RequestMapping(value = "/admin/energy", method = RequestMethod.GET)
+    public String adminEnergy(Model model,
+                              @RequestParam(value = "start", required = false) String start,
+                              @RequestParam(value = "end", required = false) String end,
+                              final RedirectAttributes redirectAttributes,
+                              HttpSession session) throws IOException {
+
+        if (!validateIfAdmin(session)) {
+            return NO_PERMISSION_PAGE;
+        }
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        ZonedDateTime now = ZonedDateTime.now();
+        if (start == null) {
+            ZonedDateTime startDate = now.with(firstDayOfMonth());
+            start = startDate.format(formatter);
+        }
+        if (end == null) {
+            ZonedDateTime endDate = now.with(lastDayOfMonth());
+            end = endDate.format(formatter);
+        }
+
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+
+        ResponseEntity responseEntity;
+        try {
+            responseEntity = restTemplate.exchange(properties.getEnergyStatistics("startDate=" + start, "endDate=" + end), HttpMethod.GET, request, String.class);
+        } catch (RestClientException e) {
+            log.warn("Error connecting to sio team service for energy usage: {}", e);
+            redirectAttributes.addFlashAttribute(MESSAGE, ERR_SERVER_OVERLOAD);
+            return "redirect:/energy_usage";
+        }
+
+        String responseBody = responseEntity.getBody().toString();
+        JSONArray jsonArray = new JSONArray(responseBody);
+
+        // handling exceptions from SIO
+        if (RestUtil.isError(responseEntity.getStatusCode())) {
+            MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
+            ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+            switch (exceptionState) {
+                case START_DATE_AFTER_END_DATE_EXCEPTION:
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "start". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1226)